File: Java/ReverseString.java
class ReverseString { //change the class name to Solution
    public void reverseString(char[] s) {
        reverse(s, 0, s.length-1);
    }

    public void reverse(char[] s, int left, int right) {
        if(left >= right) {
            return;
        }

        char aux = s[left];
        s[left] = s[right];
        s[right] = aux;
        reverse(s, left+1, right-1);
    }
}

File: C++/reverseString.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        reverse(s, 0, s.size() - 1);
    }
    void reverse(vector<char>& s, int start, int end)
    {
        if (start >= end)
        {
            return;
        }

        char aux = s[start];
        s[start] = s[end];
        s[end] = aux;

        reverse(s, start + 1, end - 1);
    }
};
